packages/babel-plugin/src/styled/__tests__/object-literal.test.tsx

```diff
@@ -373,4 +373,26 @@ describe('styled component object literal', () => {
 
     expect(actual).toInclude('{color:red}');
   });
+
+  it('should handle destructuring in interpolation functions', () => {
+    const actual = transform(
+      `
+      import { styled } from '@compiled/react';
+      import colors from 'colors';
+
+      export const BadgeSkeleton = styled.span({
+        backgroundColor: ({ isLoading }) => isLoading ? colors.N20 : colors.N40,
+        color: ({ loading: l }) => l ? colors.N50 : colors.N10,
+        borderColor: (propz) => propz.loading ? colors.N100 : colors.N200,
+      });
+    `
+    );
+
+    expect(actual).toIncludeMultiple([
+      '{as:C="span",style,isLoading,loading,...props}',
+      'isLoading?colors.N20:colors.N40',
+      'loading?colors.N50:colors.N10',
```

Comment:
hmm i think this is smart (instead of using `l` you use `loading`) - because it gets around the potential of naming collisions

can we have one test that asserts that so the knowledge is passed onto the next developer?

Reply:
🎉 yups sure